### PR TITLE
Fix missing trigger from meta block epoch start

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed missing trigger from epoch start meta block

### CODE REVIEW

File: process/block/shardblock_test.go [hunk 1]
```diff
@@ -4558,3 +4558,64 @@ func TestShardProcessor_RequestMetaHeadersIfNeededShouldAddHeaderIntoTrackerPool
...
+	dataComponents.BlockChain = blockChainMock
+	dataComponents.DataPool = poolsHolderStub
+	arguments := CreateMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+	arguments.EpochStartTrigger = epochStartTriggerStub
+	arguments.RequestHandler = requestHandlerStub
+	sp, _ := blproc.NewShardProcessor(arguments)
+
+	err := sp.CheckEpochCorrectness(nextHeader)
+
+	time.Sleep(100 * time.Millisecond)
```
Comment (iulianpascalau) [inline process/block/shardblock_test.go:4616]: can use a channel here:
On L4597 you declare a channel:
```GO
ch := make(chan struct{})
```

On L4601 you close the dummy channel:
```GO
close(ch)
```

Here, on L4616 you can have:
```GO
select{
case <- ch:
case <- time.After(time.Minute):
    assert.Fail(t, "timeout while waiting the sending of the request for the meta header")
}
```

Comment (SebastianMarian) [reply process/block/shardblock_test.go]: Ok, done

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: process/block/shardblock.go [hunk 2]
```diff
@@ -378,6 +378,10 @@ func (sp *shardProcessor) checkEpochCorrectness(
 	incorrectStartOfEpochBlock := header.GetEpoch() != currentBlockHeader.GetEpoch() &&
 		sp.epochStartTrigger.MetaEpoch() == currentBlockHeader.GetEpoch()
 	if incorrectStartOfEpochBlock {
+		if header.IsStartOfEpochBlock() {
```
Comment (sasurobert) [inline process/block/shardblock.go:381]: I gave an approve. But still thinking if this opens up other problems or not. Cannot say for sure know. But can't confirm it either, yet.

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
image

**Pull request merged**